marquez_client/client.py

```diff
@@ -9,243 +9,261 @@
         if not namespace_name:
             namespace_name = self._namespace_name
 
-        path = "namespaces/{0}/jobs/"
-        path_args = [namespace_name]
-        result = self.get_request(path, path_args)
-        return result
+        return self._get(
+            self._url('/namespaces/{0}/jobs', namespace_name),
+            params={
+                'limit': limit,
+                'offset': offset
+            })
+
+    def create_job_run(self, job_name, nominal_start_time=None, nominal_end_time=None, run_args=None, mark_as_running=False, namespace_name=None):
+        if not job_name:
+            raise ValueError('job_name must not be None')
 
-    @handle_response
-    def create_job_run(self, job_name, job_run_args=None, namespace_name=None,
-                       nominal_start_time=None,
-                       nominal_end_time=None):
         if not namespace_name:
             namespace_name = self._namespace_name
 
-        path = "namespaces/{0}/jobs/{1}/runs"
-        path_args = [namespace_name, job_name]
-        payload = {
-            "nominalStartTime": nominal_start_time,
-            "nominalEndTime": nominal_end_time,
-            "runArgs": job_run_args
-        }
-        return self.post_request(path, path_args, payload)
-
-    @handle_response
-    def get_job_run(self, job_run_id):
-        path = "jobs/runs/{0}"
-        path_args = [job_run_id]
-        return self.get_request(path, path_args)
-
-    def mark_job_run_as_running(self, job_run_id):
-        return self._mark_job_run_as(
-            job_run_id, "run")
-
-    def mark_job_run_as_completed(self, job_run_id):
-        return self._mark_job_run_as(
-            job_run_id, "complete")
-
-    def mark_job_run_as_failed(self, job_run_id):
-        return self._mark_job_run_as(
-            job_run_id, "fail")
-
-    def mark_job_run_as_aborted(self, job_run_id):
-        return self._mark_job_run_as(
-            job_run_id, "abort")
-
-    @handle_response
-    def _mark_job_run_as(self, job_run_id, action):
-        path = "jobs/runs/{0}/{1}".format(job_run_id, action)
-        path_args = [job_run_id, action]
-        payload = None
-        return self.put_request(path, path_args, payload)
-
-    #  Datasets API
-    @handle_response
-    def create_dataset(self, name, datasource_urn, namespace_name=None,
-                       description=None):
+        response = self._post(
+            self._url('/namespaces/{0}/jobs/{1}/runs',
+                      namespace_name, job_name),
+            payload={
+                'nominalStartTime': nominal_start_time,
+                'nominalEndTime': nominal_end_time,
+                'runArgs': run_args
+            })
+
+        if mark_as_running:
+            run_id = response['runId']
+            response = self.mark_job_run_as_running(run_id)
+
+        return response
+
+    def get_job_run(self, run_id):
+        return self._get(self._url('/jobs/runs/{0}', run_id))
```

Comment:
If we're checking for `run_id` being `None` in `_mark_job_run_as`, we should also check it here for consistency.

Reply:
Missed this, thanks!